nixos/modules/services/printing/cupsd.nix

```diff
@@ -384,14 +384,11 @@ in
         preStart = lib.optionalString cfg.stateless ''
           rm -rf /var/cache/cups /var/lib/cups /var/spool/cups
         '' + ''
-            mkdir -m 0700 -p /var/cache/cups
-            mkdir -m 0700 -p /var/spool/cups
-            mkdir -m 0755 -p ${cfg.tempDir}
-
-            mkdir -m 0755 -p /var/lib/cups
+            (umask 077 && mkdir -p /var/cache/cups /var/spool/cups)
```

Comment:
I just realized that this can e.g. also create `/var/cache` with perms 700, whereas it probably should have 755 (at least my `/var/cache` has that).

What's the consequence of that?

Looking at my system:

```
$ ls -l /var/cache/
total 107632
drwx------ 3 root          root              4096 Jan 19  2024 apparmor
drwxrwx--- 3 root          lp                4096 Nov 26 17:21 cups
drwxr-xr-x 3 root          root              4096 Nov 27 08:34 fwupd
drwxr-xr-x 2 fwupd-refresh fwupd-refresh     4096 Dec 28  2023 fwupdmgr
...
```

I think this means that if cups creates `/var/cache` with 700 perms, fwupd might fail to use `/var/cache/fwupdmgr`.

Should we revert?`

Reply:
FWIW, `/var/cache` and `/var/spool` both have 755 permissions on my system.

Considering this in a wider context, are we missing some centralised management of shared directories in nixpkgs? Because we have heaps of instances of `mkdir -p`, where presumably whichever runs first "wins".

Reply:
I think part of this problem is solved by systemd.tmpfiles, but yes, I'm not aware of a truly central directory creation system. 

Let's revert for now?

Reply:
>  I wish mkdir had separate options for "create the entire directory stack if it doesn't exist" and "create only the leaf directory if it doesn't exist".

Isn't that exactly `mkdir -p` and `mkdir` respectively? Or did I misunderstand what you meant?

Reply:
No, that's missing the "if it doesn't exist" part - `mkdir PATH` will *fail* if `PATH` exists.

Reply:
> I think part of this problem is solved by systemd.tmpfiles, but yes, I'm not aware of a truly central directory creation system.
> 
> Let's revert for now?

Or we can move forward by adding `(umask 022 && mkdir -p /var/cache /var/spool)`?